CHANGELOG.md

```diff
@@ -77,6 +77,9 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### New features
 
+* `jj undo` now shows a hint when undoing an undo operation that the user may
+   be looking for `jj op restore` instead.
+
```

Comment:
Don't forget to move this to "[Unreleased]". I actually don't think we even need to mention it. Users are not going to need to know about it before they run into it.

Reply:
I've moved the message, but left it in, because I think it's valuable to point out these minor UX changes.  Yes, most users won't notice it until they need it, but I think it's helpful for someone who's already run into this issue to be able to read the changelog and see that if they recommend JJ to someone else, that person _won't_ run into this issue.